src/commands/ban_commands.py

```diff
@@ -21,10 +21,21 @@ async def ban(
         reason: str,
     ):
         """Ban the specified user."""
+
         async with create_response_context(interaction) as response_message:
-            async with create_logging_embed(
-                interaction, user=user, reason=reason
-            ) as logging_embed:
-                await ban_user(logging_embed, user, reason)
+            (ban_state, dm_state, result_description) = await ban_user(user, reason)
 
-                response_message.set_string(f"Successfully banned {user.mention}")
+            if ban_state:  # ban succeeded
+                if not dm_state:  # dm failed
```

Comment:
~nit: with this naming convention, i think it reads better if we do `if dm_state is None`~. looking further down the code i see that this is a bool, bools should be typically named as `is_xxx` to make it more clear that it is a bool and not an object (for both `ban_state` and `dm_state`)